### ISSUE TITLE
[Feature request] no-registered-classes rule

### ISSUE THREAD

**Issue by jamiebuilds-signal** [schoero/eslint-plugin-readable-tailwind#209]
As part of migrating from a large Sass codebase to Tailwind we wanted to make Sass/Tailwind class names mutually exclusive, so we created the inverse rule of `no-unregistered-classes`

```tsx
<div className="legacy-sass-style p-4"/>
//                                ^^^
//               BAD: Tailwind class not allowed here

<div className={classNames("legacy-sass-style p-4")}/>
//                                            ^^^
//               BAD: Tailwind class not allowed here

<div className={tw("legacy-sass-style p-4")}/>
//                  ^^^^^^^^^^^^^^^^^ 
//               BAD: Only Tailwind classes allowed here
```

Here is the source of that rule although it only does enough to work in our codebase:

https://gist.github.com/jamiebuilds-signal/660b2f5a8d2e6895cf3ffc803b5e2350

I'm imagining it being configurable like this:

```tsx
{
  "settings": {
    "better-tailwindcss": {
      "entryPoint": "src/tailwind.css",
      "callees": ["tw"],
      "attributes": [],
      "variables": [],
    },
  },
  "rules": {
    "better-tailwindcss/no-unregistered-classes": "error",
    "better-tailwindcss/no-registered-classes": ["error", {
      "callees": ["classNames"],
      "attributes": ["className"],
      "variables": ["^classNames?$"],
    }],
  }
}
```
